src/mono/wasm/templates/templates/browser/app-support.js

```diff
@@ -149,44 +147,43 @@ function applyArguments() {
     }
 }
 
-let toAbsoluteUrl = function(possiblyRelativeUrl) { return possiblyRelativeUrl; }
+let toAbsoluteUrl = function (path, prefix) {
+    if (prefix.startsWith("/")) {
+        return path;
+    }
+    return prefix + path;
+}
 const anchorTagForAbsoluteUrlConversions = document.createElement('a');
-toAbsoluteUrl = function toAbsoluteUrl(possiblyRelativeUrl) {
-   anchorTagForAbsoluteUrlConversions.href = possiblyRelativeUrl;
-   return anchorTagForAbsoluteUrlConversions.href;
+toAbsoluteUrl = function toAbsoluteUrl(path, prefix) {
+    anchorTagForAbsoluteUrlConversions.href = prefix + path;
+    return anchorTagForAbsoluteUrlConversions.href;
 }
 
-let loadDotnetPromise = import('/dotnet.js');
-let argsPromise = fetch('/runArgs.json')
-                    .then(async (response) => {
-                        if (!response.ok) {
-                            console.debug(`could not load /args.json: ${response.status}. Ignoring`);
-                        } else {
-                            runArgs = await response.json();
-                            console.debug(`runArgs: ${JSON.stringify(runArgs)}`);
-                        }
-                    })
-                    .catch(error => console.error(`Failed to load args: ${stringify_as_error_with_stack(error)}`));
-
-Promise.all([ argsPromise, loadDotnetPromise ]).then(async ([ _, { default: createDotnetRuntime } ]) => {
+try {
+    initRunArgs();
+    const argsResponse = await fetch('./runArgs.json')
+    if (!argsResponse.ok) {
+        console.debug(`could not load ./runArgs.json: ${response.status}. Ignoring`);
+    } else {
+        runArgs = await argsResponse.json();
+        console.debug(`runArgs: ${JSON.stringify(runArgs)}`);
+    }
     applyArguments();
 
-    return createDotnetRuntime(({ MONO, INTERNAL, BINDING, Module }) => ({
+    createDotnetRuntime(({ MONO, INTERNAL, BINDING, Module }) => ({
         disableDotnet6Compatibility: true,
         config: null,
         configSrc: "./mono-config.json",
-        locateFile: (path, prefix) => {
-            return toAbsoluteUrl(prefix + path);
-        },
+        locateFile: toAbsoluteUrl,
         onConfigLoaded: (config) => {
             if (!Module.config) {
                 const err = new Error("Could not find ./mono-config.json. Cancelling run");
                 set_exit_code(1);
                 throw err;
             }
             // Have to set env vars here to enable setting MONO_LOG_LEVEL etc.
-            for (let variable in runArgs.runtimeArgs) {
-                config.environment_variables[variable] = runArgs.runtimeArgs[variable];
+            for (let variable in runArgs.environment_variables) {
+                config.environment_variables[variable] = runArgs.environment_variables[variable];
```

Comment:
`runArgs.environment_variables` -> `runArgs.environmentVariables`
The json has: `"environmentVariables":{`